### ISSUE TITLE
The time cost for the example script examples/scripts/r1_scripts/train_rloo_qwenvl2_5_math.sh

### ISSUE THREAD

**Issue by simplelifetime** [TideDra/lmm-r1#8]
I'm running the reproduction experiment on 8 A100s. It seems that one episode will already cost me more than 3 hours. Is this normal?
 
Also, during the entire experiments so far, the KL value remains 0 the whole time. I'm wondering if it's a correct phenomenon or not.

My base model is qwen2_5-VL-3B.

**Comment by TideDra** [comment]
Do you change any of the experiments setting? Your speed is not normal. It's expected to see KL = 0 because we do not use kl loss so there is no ref model and kl is not calculated.

**Comment by simplelifetime** [comment]
No, I didn't change any except deleting --runtime-env-json. Below is my evaluation script:
ray job submit --address="http://127.0.0.1:8265" \
   -- python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 8 \
   --remote_rm_url http://127.0.0.1:5000/get_reward \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 8 \
   --vllm_num_engines 8 \
   --vllm_tensor_parallel_size 1 \
   --colocate_all_models \
   --vllm_enable_sleep \
   --vllm_gpu_memory_utilization 0.7 \
   --vllm_sync_backend gloo \
   --enable_prefix_caching \
   --pretrain $PRETRAIN_MODEL \
   --save_path $SAVE_PATH/$MODEL_CPK_NAME \
   --micro_train_batch_size 2 \
   --train_batch_size 128 \
   --micro_rollout_batch_size 4 \
   --rollout_batch_size 256 \
   --temperature 1 \
   --n_samples_per_prompt 16 \
   --max_epochs 1 \
   --num_episodes 30 \
   --prompt_max_len 1024 \
   --max_samples 100000 \
   --generate_max_len 8192 \
   --advantage_estimator rloo \
   --zero_stage 3 \
   --bf16 \
   --actor_learning_rate 1e-6 \
   --init_kl_coef 0.0 \
   --prompt_data $DATASET \
   --input_key prompt \
   --normalize_reward \
   --flash_attn \
   --gradient_checkpointing \
   --save_steps 100 \
   --ckpt_path $SAVE_PATH/$MODEL_CPK_NAME/ckpt \
   --save_hf_ckpt \
   --use_tensorboard $SAVE_PATH/$MODEL_CPK_NAME/logs

Also,  the speed is approximately as below:
Train epoch [1/1]:  38%|███▊      | 97/256 [01:28<02:19,  1.14it/s, pg=-0.297, rm=0.781, ret=0.0958, glen=560, tlen=753

**Comment by TideDra** [comment]
This script is tested on 8*A6000 (49G). You can increase the micro batchsize for A100. The training speed seems to be 4min/step and is ok under the default setting. What's the speed of making experience and sampling?  The total time for one step should be around 10 min. One episode takes 8000/256 = 31 step, so you just need 7 hours to get our 80 steps results.

**Comment by simplelifetime** [comment]
Thanks! That seems about right. I have another question about the kl value. I notice that kl value presented in the readme tensorboard file is not always zero. However, it seems not the case for me as my value is zero through the 18 steps I've trained. I don't know if it will become a problem.

Here are the two tensorboard results (below is mine). Note that I haven't finished training yet.

![Image](https://github.com/user-attachments/assets/9867480a-c93b-4310-86a4-540806882697)

![Image](https://github.com/user-attachments/assets/a007ee60-7c10-433f-8d3e-e1f3d86449a0)

**Comment by TideDra** [comment]
it's ok. The code is updated since we release the experiments report. The new code cancel ref model to save gpu memory and no kl will be calculate if kl coef=0.

**Comment by simplelifetime** [comment]
Thanks a lot! That solves my problem!
